### DISCUSSION TITLE
Should Netty's TimerTask has a `cancel/release` method?

### DISCUSSION THREAD

**Discussion by He-Pin** [netty/netty#14566] [General]
When the Timer is stopped, the timeout is only returned and then a user needs to do some release. Eg, the `TimerTask#run` block may closing over a `ByteBuf`, but still a user can cancel all the timeout and then run them all to do the release too, not sure if separating them is a good idea.

**Comment by normanmaurer** [comment]
@He-Pin sounds like a good idea https://github.com/netty/netty/pull/14571 PTAL

**Reply by He-Pin** [reply]
Thanks, because we are investigating a leak in AMDC, our traffic dispatcher.
